src/Sulu/Bundle/ContactBundle/Content/Types/SingleAccountSelection.php

```diff
@@ -36,19 +37,59 @@ public function __construct(
     ) {
         $this->accountManager = $accountManager;
         $this->accountReferenceStore = $accountReferenceStore;
+    }
+
+    public function read(NodeInterface $node, PropertyInterface $property, $webspaceKey, $languageCode, $segmentKey)
+    {
+        $account = null;
```

Comment:
There should be more error checking if the account really exist. And read currently does always return the value so I would do this:

```
        $value = null;

        if ($node->hasProperty($property->getName())) {
            $account = $this->accountManager->getById(
                $node->getPropertyValue($property->getName()),
                $property->getStructure()->getLanguageCode()
            );

            if ($account) {
                $value = [
                    'id' => $account->getId(),
                    'name' => $account->getName(),
                ];
            }
        }

        $property->setValue($value);

        return $value;
```

Reply:
Not all `read` methods return a value, so I assumed that the value is not used by the caller at all... And if this is the case I think it is really confusing to return something...

And the other thing is that an exception is thrown if the account is not existing... But gonna fix it anyway.